src/Metadata/DateTimeOptions.php

```diff
@@ -20,17 +20,22 @@ final class DateTimeOptions implements \JsonSerializable
     private $zone;
 
     /**
-     * Use if a different format should be used for parsing dates than for generating dates.
+     * Use if different formats should be used for parsing dates than for generating dates.
      *
-     * @var string|null
+     * @var string[]|null
      */
-    private $deserializeFormat;
+    private $deserializeFormats;
 
-    public function __construct(?string $format, ?string $zone, ?string $deserializeFormat)
+    /**
+     * @note Passing a string for $deserializeFormats is deprecated, please pass an array instead
+     *
+     * @param string[]|string|null $deserializeFormats
+     */
+    public function __construct(?string $format, ?string $zone, $deserializeFormats = null)
```

Comment:
the constructor argument was not optional before, we should not make it optional now. it was (and still is) nullable.

```suggestion
    public function __construct(?string $format, ?string $zone, $deserializeFormats)
```